src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -112,7 +123,9 @@ protected void doXContent(XContentBuilder xContentBuilder, Params params) throws
         xContentBuilder.startObject(NAME);
         xContentBuilder.startObject(fieldName);
         xContentBuilder.field(QUERY_TEXT_FIELD.getPreferredName(), queryText);
-        xContentBuilder.field(MODEL_ID_FIELD.getPreferredName(), modelId);
+        if (!isClusterOnOrAfterMinRequiredVersion() || (isClusterOnOrAfterMinRequiredVersion() && modelId != null)) {
```

Comment:
Can this be simplified to:
```
if (modelId != null) {

}
```

Reply:
Yes, I was planning to do the same. But @navneet1v earlier suggested to put this clusterOnorAfter checks so to maintain the similarity in the work I used this one. Any thoughts @navneet1v ?